bundles/nl.asml.matala.product/src/nl/asml/matala/product/generator/ValidationHelper.java

```diff
@@ -0,0 +1,70 @@
+package nl.asml.matala.product.generator;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+
+
+import nl.esi.comma.actions.actions.Action;
+import nl.esi.comma.actions.actions.AssignmentAction;
+import nl.esi.comma.actions.actions.ForAction;
+import nl.esi.comma.actions.actions.IfAction;
+import nl.esi.comma.actions.actions.RecordFieldAssignmentAction;
+import nl.esi.comma.expressions.expression.Expression;
+import nl.esi.comma.expressions.expression.ExpressionRecordAccess;
+import nl.esi.comma.expressions.generator.ExpressionsCommaGenerator;
+
+public class ValidationHelper {
+	
+	/**
+	 * Get variables based on the action type
+	 * */
+	
+	public static Map<String, Object> action(Action action) {
+		if (action instanceof AssignmentAction) {
+			AssignmentAction a = (AssignmentAction) action;
```

Comment:
You don't need this explicit cast in Xtend, the if will guarantee action to be an `AssignmentAction`, just try it without casting.

Reply:
Ah, I see now You can either make this an xtend class too, that will ease its implementation. Or in Java you can do:
```java
if (action instanceof AssignmentAction a) {
```